src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimSpecCallbackHandler.java

```diff
@@ -929,36 +927,45 @@ private CallbackResponse validateSpecRespondent2RepEmail(CallbackParams callback
             .build();
     }
 
-    private CallbackResponse isFlightDelayClaim(CallbackParams callbackParams) {
+    private CallbackResponse getAirlineList(CallbackParams callbackParams) {
         CaseData.CaseDataBuilder<?, ?> caseDataBuilder = callbackParams.getCaseData().toBuilder();
-
         if (toggleService.isSdoR2Enabled()) {
-            caseDataBuilder.isFlightDelayClaim(callbackParams.getCaseData().getIsFlightDelayClaim());
-            if (callbackParams.getCaseData().getIsFlightDelayClaim().equals(YES)) {
-                caseDataBuilder.claimType(ClaimType.FLIGHT_DELAY);
-            } else {
-                caseDataBuilder.claimType(null);
-            }
-        }
+            List<AirlineEpimsId> airlineEpimsIDList = new ArrayList<>(airlineEpimsDataLoader.getAirlineEpimsIDList());
+            DynamicList airlineList = DynamicList
+                .fromList(
+                    airlineEpimsIDList.stream()
+                        .map(AirlineEpimsId::getAirline).toList(),
+                    Object::toString,
+                    Object::toString,
+                    null,
+                    false
+                );
+            DynamicList dropdownAirlineList = DynamicList.builder()
+                .listItems(airlineList.getListItems()).build();
 
+            FlightDelayDetails flightDelayDetails = FlightDelayDetails.builder().airlineList(dropdownAirlineList).build();
+            caseDataBuilder.flightDelayDetails(flightDelayDetails);
+        }
         return AboutToStartOrSubmitCallbackResponse.builder()
             .data(caseDataBuilder.build().toMap(objectMapper))
             .build();
     }
 
-    private CallbackResponse getAirlineList(CallbackParams callbackParams) {
+    private CallbackResponse validateDateAndSetFlightDelayClaimType(CallbackParams callbackParams) {
```

Comment:
This callback should be renamed to "validateDate". We are not setting the claimType here.

Reply:
Renamed to validateFlightDelayDate